data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -112,59 +115,65 @@ public KafkaSource(final KafkaSourceConfig sourceConfig,
-        KafkaSecurityConfigurer.setAuthProperties(authProperties, sourceConfig, LOG);
-        sourceConfig.getTopics().forEach(topic -> {
-            consumerGroupID = topic.getGroupId();
-            KafkaTopicConsumerMetrics topicMetrics = new KafkaTopicConsumerMetrics(topic.getName(), pluginMetrics, true);
-            Properties consumerProperties = getConsumerProperties(topic, authProperties);
-            MessageFormat schema = MessageFormat.getByMessageFormatByName(schemaType);
-            try {
-                int numWorkers = topic.getWorkers();
-                executorService = Executors.newFixedThreadPool(numWorkers);
-                allTopicExecutorServices.add(executorService);
-
-                IntStream.range(0, numWorkers).forEach(index -> {
-                    while (true) {
-                        try {
-                            kafkaConsumer = createKafkaConsumer(schema, consumerProperties);
-                            break;
-                        } catch (ConfigException ce) {
-                            if (ce.getMessage().contains(NO_RESOLVABLE_URLS_ERROR_MESSAGE)) {
-                                LOG.warn("Exception while creating Kafka consumer: ", ce);
-                                LOG.warn("Bootstrap URL could not be resolved. Retrying in {} ms...", RETRY_SLEEP_INTERVAL);
-                                try {
-                                    sleep(RETRY_SLEEP_INTERVAL);
-                                } catch (InterruptedException ie) {
-                                    Thread.currentThread().interrupt();
-                                    throw new RuntimeException(ie);
+        try {
+            setMdc();
+            Properties authProperties = new Properties();
+            KafkaSecurityConfigurer.setDynamicSaslClientCallbackHandler(authProperties, sourceConfig, pluginConfigObservable);
+            KafkaSecurityConfigurer.setAuthProperties(authProperties, sourceConfig, LOG);
+            sourceConfig.getTopics().forEach(topic -> {
+                consumerGroupID = topic.getGroupId();
+                KafkaTopicConsumerMetrics topicMetrics = new KafkaTopicConsumerMetrics(topic.getName(), pluginMetrics, true);
+                Properties consumerProperties = getConsumerProperties(topic, authProperties);
+                MessageFormat schema = MessageFormat.getByMessageFormatByName(schemaType);
+                try {
+                    int numWorkers = topic.getWorkers();
+                    final ExecutorService executorService = Executors.newFixedThreadPool(
+                            numWorkers, KafkaPluginThreadFactory.defaultExecutorThreadFactory(MDC_KAFKA_PLUGIN_VALUE, topic.getName()));
+                    allTopicExecutorServices.add(executorService);
+
+                    IntStream.range(0, numWorkers).forEach(index -> {
+                        while (true) {
```

Comment:
This part of the code is increasing, Probably good idea to move to separate helper function. Maybe a separate PR.